### PR TITLE
feat(halo/evmstaking2): unit test for validator creation event

### PR DESCRIPTION
**Pull request by chmllr**
Extends the unit test with delivery of validator creation event.

issue: #2525

### CODE REVIEW

File: halo/evmstaking2/keeper/keeper_internal_test.go [hunk 1]
```diff
@@ -165,10 +171,22 @@ func setupKeeper(
...
+// on the pubkey of the new validator.
+func (m *stakingKeeperMock) GetValidator(context.Context, sdk.ValAddress) (stypes.Validator, error) {
+	if m.calls == 0 {
+		m.calls++
+		return stypes.Validator{}, nil
+	}
 
-func (stakingKeeperMock) GetValidator(context.Context, sdk.ValAddress) (stypes.Validator, error) {
-	return stypes.Validator{}, nil
+	return stypes.Validator{}, errors.New("validator exists")
```
Comment (corverroos) [inline halo/evmstaking2/keeper/keeper_internal_test.go:189]: mmmm, usually errors are returned if the validator does NOT exist...?

File: halo/evmstaking2/keeper/keeper_internal_test.go [hunk 2]
```diff
@@ -165,10 +171,22 @@ func setupKeeper(
 	return k, ctx
 }
 
-type stakingKeeperMock struct{}
+type stakingKeeperMock struct {
+	// calls is the number of calls to GetValidator
+	calls uint32
```
Comment (corverroos) [inline halo/evmstaking2/keeper/keeper_internal_test.go:176]: this will only ever be 0 or 1, and isn't technically correct since additional calls to GetValidator are not counted

File: halo/evmstaking2/keeper/keeper_internal_test.go [hunk 3]
```diff
@@ -90,48 +91,53 @@ func TestInsertAndDeleteEVMEvents(t *testing.T) {
...
-	// Make sure the event was persisted.
-	insertedID := uint64(1)
-	found, err := keeper.eventsTable.Has(ctx, insertedID)
-	require.NoError(t, err)
-	require.True(t, found)
+	// Make sure the events were persisted.
+	for id := uint64(1); id < 3; id++ {
+		found, err := keeper.eventsTable.Has(ctx, id)
+		require.NoError(t, err)
+		require.True(t, found)
```
Comment (corverroos) [inline halo/evmstaking2/keeper/keeper_internal_test.go:124]: feels like a helper function would help here, since we do this many times in all the tests:

```
assertContains(t, keeper, id)
assertNotContains(t, keeper, id)
```

This helps make tests much more readable.

File: halo/evmstaking2/keeper/keeper_internal_test.go [hunk 4]
```diff
@@ -90,48 +91,53 @@ func TestInsertAndDeleteEVMEvents(t *testing.T) {
...
 		require.NoError(t, err)
 	}
 
-	// Make sure the event was persisted.
-	insertedID := uint64(1)
-	found, err := keeper.eventsTable.Has(ctx, insertedID)
-	require.NoError(t, err)
-	require.True(t, found)
+	// Make sure the events were persisted.
+	for id := uint64(1); id < 3; id++ {
```
Comment (corverroos) [inline halo/evmstaking2/keeper/keeper_internal_test.go:121]: magic numbers are a bit confusing. Lets create a variables rather?
```
expectDelegates := 1
expectCreates := 1
expectTotal := expectDelegates + expectCreates
require.Len(t, events, expectTotal)
...
for id := uint64(1); id <= expectTotal; id++ {
  assertContains(t, keeper, id)
}
...
require.Len(t, msgServer.delegateMsgBuffer, expectDelegates)
require.Len(t, msgServer.createValidatorMsgBuffer, expectCreates)
```

File: halo/evmstaking2/keeper/keeper_internal_test.go [hunk 5]
```diff
@@ -90,48 +91,53 @@ func TestInsertAndDeleteEVMEvents(t *testing.T) {
 	}
 }
 
-func TestDeliverDelegate(t *testing.T) {
+func TestDelivery(t *testing.T) {
```
Comment (corverroos) [inline halo/evmstaking2/keeper/keeper_internal_test.go:94]: would be great if we can test edge cases as well, i.e. delivery errors and panics.

**Review by corverroos — APPROVED** [review]
